docs/contributing/setup.md

```diff
@@ -6,13 +6,15 @@ You will need to install these tools on your machine:
 
 ### macOS
 
- - [Node.js v8.11.2](https://nodejs.org/dist/v8.11.2/)
+ - [Node.js v8.11.4](https://nodejs.org/dist/v8.11.4/)
  - [Python 2.7](https://www.python.org/downloads/mac-osx/)
  - Xcode and Xcode Command Line Tools (Xcode -> Preferences -> Downloads)
 
+See [mac-deps-setup.md](./mac-deps-setup.md) for additional help with this step.
```

Comment:
I'm now trying to reconcile how we've duplicated the list of dependencies in two places now - is it worth just rewriting this new document as the macOS setup guide, with a summary at the top of what's needed, and then more details to walk people through the setup, outlining options?

The Windows section below is also a good candidate for it's own document rather than the complex bullet list, but that's out of scope for this PR.

Reply:
The benefit of this approach is that we can be more detailed with the setup guide, rather than trying to cram it in, and this document becomes just a list of the platforms for developing on, linking off to the specialized docs.

Reply:
yeah that's the direction i want to head. was thinking i would separate out later more cleanly, but there's no reason not to do a little clean up now

Reply:
0001615